pageserver/src/layered_repository/delta_layer.rs

```diff
@@ -526,7 +526,7 @@ impl DeltaLayer {
 
         *inner = DeltaLayerInner {
             loaded: true,
-            book: None,
+            book: Some(book),
```

Comment:
This seems like an unrelated change, but it looks confusing and I think this is correct. More context:  I've faced a panic in an unwrap call [here](https://github.com/zenithdb/zenith/blob/main/pageserver/src/layered_repository/delta_layer.rs#L206) when I was writing the unittest, and it looks strange that we never set this book to something other than None. 

@hlinnaka Is it correct?

Also I'm thinking about adding a lazy behaviour in VirtualFile, so we can call open on it but it will be actually opened on first read/write operation. Thoughts?

Reply:
@hlinnaka I'm not sure about this change, please take a look when you have a chance

Reply:
Huh, yeah, that was clearly a bug in 9300107cdfb3ee168c0ee2a553f093d8a064b2a1. Good catch!

But now I'm a bit concerned; why was this not caught by any tests? Do we not have any test coverage for `get_page_reconstruct_data` on a delta layer?

Reply:
> Also I'm thinking about adding more lazy behaviour to VirtualFile, so we can call open on it but it will be actually opened on first read/write operation. Thoughts?

Two thoughts:
1. Currently, we only open a VirtualFile if we're going to read it. So it wouldn't make any difference in practice.
2. I think it's good to check that the file exists and is accessible, on the `open` call. The simple way to do that is to actually open the file.

Reply:
> why was this not caught by any tests?

Yeah, thats why I wasnt sure about the change

> Do we not have any test coverage for get_page_reconstruct_data on a delta layer?

It seems that this is the case, I've put a panic inside get_page_reconstruct_data for delta layer and rust tests have passed (tried on main)

I've managed to get basic coverage using cargo-tarpaulin with
```
cargo install cargo-tarpaulin
cargo tarpaulin -v --avoid-cfg-tarpaulin
```
So it shows basic line coverage in the report:
```
42.35% coverage, 4418/10433 lines covered
```
It outputs line numbers for each file in the report so it is possible to see what is not covered

> Two thoughts:

Makes sense, lets keep it as it is now